flowmachine/flowmachine/core/query_state.py

```diff
@@ -26,13 +26,19 @@ class QueryState(str, Enum):
     Possible states for a query to be in.
     """
 
-    QUEUED = "queued"
-    COMPLETED = "completed"
-    CANCELLED = "cancelled"
-    EXECUTING = "executing"
-    ERRORED = "errored"
-    RESETTING = "resetting"
-    KNOWN = "known"
+    QUEUED = ("queued", "is queued.")
+    COMPLETED = ("completed", "is ready.")
+    CANCELLED = ("cancelled", "was cancelled.")
+    EXECUTING = ("executing", "is currently running.")
+    ERRORED = ("errored", "finished with an error.")
+    RESETTING = ("resetting", "is being reset.")
+    KNOWN = ("known", "is known, but has not yet been run.")
+
+    def __new__(cls, name, desc, **kwargs):
```

Comment:
That's a very clever solution. :)

The only minor downside is that the `description` attribute is not easily discoverable in an interactive IPython session because it's only added upon instance creation (so it doesn't show up in `dir(...)` and IPython can't provide tab-completion for it).

Can we add a line to the class docstring to show an example of the `value` and `description` attribute?

Reply:
Also, how about removing the period at the end of the descriptions (and putting it at the end of the `msg` string in the action handler instead)? Might make them more easily re-usable in other situations.

Reply:
I've promoted description to a property to make it more discoverable.